File: BuilderDesignPattern/CarManufacture_Example/Driver.java
package BuilderDesignPattern.CarManufacture_Example;

import BuilderDesignPattern.CarManufacture_Example.Builders.Builder;
import BuilderDesignPattern.CarManufacture_Example.Builders.CarBuilder;
import BuilderDesignPattern.CarManufacture_Example.Builders.CarManualBuilder;
import BuilderDesignPattern.CarManufacture_Example.Directors.Director;
import BuilderDesignPattern.CarManufacture_Example.Products.Car;
import BuilderDesignPattern.CarManufacture_Example.Products.CarManual;

public class Driver {
	public static void main(String[] args) {
		Builder<Car>       carBuilder       = new CarBuilder();
		Builder<CarManual> carManualBuilder = new CarManualBuilder();

		Director director = new Director();

		Car car1 = director.constructSportsCar(carBuilder);
		Car car2 = director.constructSedanCar(carBuilder);
		Car car3 = director.constructSUVCar(carBuilder);
		Car car4 = director.constructElectricCar(carBuilder);

		CarManual carManual1 = director.constructSportsCar(carManualBuilder);
		CarManual carManual2 = director.constructSedanCar(carManualBuilder);
		CarManual carManual3 = director.constructSUVCar(carManualBuilder);
		CarManual carManual4 = director.constructElectricCar(carManualBuilder);

//		System.out.println(car1.print());
//		car1.getEngine().go(200);
//		car1.getEngine().turnOff();
		System.out.println(car1.print());
		System.out.println(car2.print());
		System.out.println(car3.print());
		System.out.println(car4.print());

		System.out.println(carManual1.print());
		System.out.println(carManual2.print());
		System.out.println(carManual3.print());
		System.out.println(carManual4.print());
	}
}

File: BuilderDesignPattern/CarManufacture_Example_Python/Driver.py
from BuilderDesignPattern.CarManufacture_Example_Python.Builders.CarBuilder import CarBuilder
from BuilderDesignPattern.CarManufacture_Example_Python.Builders.CarManualBuilder import CarManualBuilder
from BuilderDesignPattern.CarManufacture_Example_Python.Directors.CarDirector import CarDirector
from BuilderDesignPattern.CarManufacture_Example_Python.Products.Car import Car
from BuilderDesignPattern.CarManufacture_Example_Python.Products.CarManual import CarManual
from BuilderDesignPattern.CarManufacture_Example_Python.Components.Engine import Engine
from BuilderDesignPattern.CarManufacture_Example_Python.Components.TripComputer import TripComputer
from BuilderDesignPattern.CarManufacture_Example_Python.Components.Transmission import Transmission
from BuilderDesignPattern.CarManufacture_Example_Python.Components.GPSNavigator import GPSNavigator
from BuilderDesignPattern.CarManufacture_Example_Python.ProductsTypes.CarsType import CarsType


class Driver:
	@staticmethod
	def main():

		director: CarDirector = CarDirector()
		carBuilder: CarBuilder = CarBuilder()
		customCarBuilder: CarBuilder = CarBuilder()
		carManualBuilder: CarManualBuilder = CarManualBuilder()

		car1: Car = director.constructSportsCar(carBuilder)
		car2: Car = director.constructElectricCar(carBuilder)
		car3: Car = director.constructSedanCar(carBuilder)
		car4: Car = director.constructSUVCar(carBuilder)
		car5: Car = (customCarBuilder.setCarModel("HONDA Accord")
					 .setCarType(CarsType.HYBRID)
					 .buildSeats(4)
					 .buildEngine(Engine(2500, 47))
					 .buildTransmission(Transmission.AUTOMATIC)
					 .buildTripComputer(TripComputer())
					 .buildGPSNavigator(GPSNavigator())
					 .build())

		carManual1: CarManual = director.constructSportsCar(carManualBuilder)

		print(car1)
		print(car2)
		print(car3)
		print(car4)
		print(car5)

		print(carManual1)

if __name__ == "__main__":
	Driver.main()